application/src/main/java/org/togetherjava/tjbot/features/help/MetadataMiner.java

```diff
@@ -0,0 +1,123 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.features.EventReceiver;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import static org.togetherjava.tjbot.db.generated.Tables.HELP_THREADS;
+
+public class MetadataMiner extends ListenerAdapter implements EventReceiver {
+    private final Database database;
+    private final Predicate<String> isHelpForumName;
+    private static final Logger logger = LoggerFactory.getLogger(MetadataMiner.class);
+    private final Cache<Long, Instant> threadIdToCreatedAtCache = Caffeine.newBuilder()
+        .maximumSize(1_000)
+        .expireAfterAccess(2, TimeUnit.of(ChronoUnit.MINUTES))
+        .build();
+
+
+    public MetadataMiner(Database database, Config config) {
+        this.database = database;
+        isHelpForumName =
+                Pattern.compile(config.getHelpSystem().getHelpForumPattern()).asMatchPredicate();
+    }
+
+    @Override
+    public void onChannelCreate(@NotNull ChannelCreateEvent event) {
+        if (!event.getChannelType().isThread()) {
+            return;
+        }
+
+        ThreadChannel threadChannel = event.getChannel().asThreadChannel();
+        String rootName = threadChannel.getParentChannel().getName();
+
+        if (isHelpForumName.test(rootName)) {
+            handleHelpThreadCreated(event, threadChannel);
+        }
+    }
+
+    @Override
+    public void onChannelUpdateArchived(@NotNull ChannelUpdateArchivedEvent event) {
+        if (!event.getChannelType().isThread()
+                || !event.getChannel().asThreadChannel().isArchived()) {
+            return;
+        }
+
+        ThreadChannel threadChannel = event.getChannel().asThreadChannel();
+        String rootName = threadChannel.getParentChannel().getName();
+
+        if (isHelpForumName.test(rootName)) {
+            handleHelpThreadClosed(threadChannel);
+        }
+    }
+
+    private void handleHelpThreadCreated(ChannelCreateEvent event, ThreadChannel threadChannel) {
+        if (wasThreadAlreadyHandled(threadChannel.getIdLong())) {
+            return;
+        }
+
+        String tag = threadChannel.getAppliedTags().get(0).getName();
+        long channelId = threadChannel.getIdLong();
+        Instant createdAt = Instant.now();
+
+        //ACTUAL AUTHOR SHOULD BE REGISTERED IF TRANSFERRED
+        long authorId = threadChannel.getOwnerIdLong();
+
+        logger.warn("ticket tags applied: "+tag);
+
+            database.write(context -> context.newRecord(HELP_THREADS)
+                .setChannelId(channelId)
+                .setCreatedAt(createdAt)
+                .setAuthorId(authorId)
+                .setTicketStatus(TicketStatus.ACTIVE.val)
+                .setTag(tag)
+                .insert());
```

Comment:
im struggling with this bit, somehow it's doing writes twice. Error pops up with integrity violation for duplicate `threadId`.
I can't find what's going on though, thoughts?